llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1238,7 +1239,8 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
             // If we are loading a full row major matrix, need to load the rows and then transpose.
             for (unsigned i = 0; i < rowCount; i++) {
               Value *pointerElem = getBuilder()->CreateGEP(
-                  pointer, {getBuilder()->getInt32(0), getBuilder()->getInt32(i), getBuilder()->getInt32(0)});
+                  pointer->getType()->getScalarType()->getPointerElementType(), pointer,
```

Comment:
Could we clean up the use of `pointer->getType()` in this function?
I can see it used in:
*  `if (!pointer->getType()->isPointerTy()) {`
*  `Type *const pointerType = pointer->getType();`
*  `pointer->getType()->getScalarType()->getPointerElementType()`